vm.h

```diff
@@ -46,4 +48,22 @@ void vm_hook_opcode_handler(vm_env *, int, vm_handler);
 
 void vm_run(vm_env *env);
 
+vm_seg_info *vm_new_seg_info(char *mem, size_t sz);
```

Comment:
We can follow [the manner of LLVM](http://blog.llvm.org/2010/04/intro-to-llvm-mc-project.html) and create new header like `mc.h` or `mach-code.h`, which enables the generation of machine code.